crypto/src/main/java/org/springframework/security/crypto/keys/StaticKeyGeneratingKeyManager.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.crypto.keys;
+
+import org.springframework.util.Assert;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *
+ * @author Joe Grandja
+ * @since 0.0.1
+ */
+public final class StaticKeyGeneratingKeyManager implements KeyManager {
```

Comment:
I'd call this `DefaultKeyManager` and document that it generates HMAC and RSA keys. Or alternatively make the generated keys configurable.

Reply:
This implementation is strictly for development/testing and I've javadoc'd it. This is good for now but it will likely go away later on when a new implementation is provided.